src/Groups/directproducts.jl

```diff
@@ -171,25 +171,36 @@ DirectProduct of
  Sym( [ 1 .. 3 ] )
  Sym( [ 1 .. 2 ] )
 
-julia> as_perm_group(G)
+julia> permutation_group(G)
 Group([ (1,2,3), (1,2), (4,5) ])
 ```
 """
-function as_perm_group(G::DirectProductGroup)
+function permutation_group(G::DirectProductGroup)
```

Comment:
So far we used `PermGroup(g)` to convert a given group `g` into a `PermGroup`, quite intentionally using the group name.

I'd also be fine with switching that to e.g. `permutation_group`, but then it should be done uniformly. Right now, with this PR we'd be in a situation were converting a PcGroup to a PermGroup works via `PermGroup`, while converting a direct product group to a `PermGroup` would require `permutation_group`. That's very confusing.

Note that the existing `permutation_group` command does something quite different (it takes a degree and a bunch of permutations). Oh, and then there are `permutation_group` methods in `experimental` which are even different... let's ignore those..

(The same of course also applies to `pc_group` versus `PcGroup`.)



*independent* of that, it actually seems these conversions already work in `master` thanks to this generic method:
```
"""
    FPGroup(G::T) where T <: Union{GAPGroup, GrpAbFinGen}
    GrpAbFinGen(G::T) where T <: GAPGroup
    PcGroup(G::T) where T <: Union{GAPGroup, GrpAbFinGen}
    PermGroup(G::T) where T <: Union{GAPGroup, GrpAbFinGen}

Return a group of type `T` that is isomorphic with `G`.
If one needs the isomorphism then
[isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup, PermGroup}](@ref)
can be used instead.
"""
function (::Type{S})(G::T) where {S <: Union{GrpAbFinGen, GAPGroup}, T <: GAPGroup}
   return codomain(isomorphism(S, G))
end
```
So indeed:
```
julia> H = symmetric_group(3)
Sym( [ 1 .. 3 ] )

julia> K = symmetric_group(2)
Sym( [ 1 .. 2 ] )

julia> G = direct_product(H,K)
DirectProduct of
 Sym( [ 1 .. 3 ] )
 Sym( [ 1 .. 2 ] )

julia> PermGroup(G)
Group([ (1,2,3), (1,2), (4,5) ])

julia> PcGroup(G)
Group([ f1, f2, f3 ])

julia> isomorphism(PermGroup, G)
Group homomorphism from
DirectProduct of
 Sym( [ 1 .. 3 ] )
 Sym( [ 1 .. 2 ] )
to
Group([ (1,2,3), (1,2), (4,5) ])
```


In light of this, perhaps we should just remove `as_perm_group` and `as_polycyclic_group`. And then possibly open an issue about adding `permutation_group` / `pc_group` aliases?